haproxy_exporter.go

```diff
@@ -352,6 +375,27 @@ loop:
 	return 1
 }
 
+func (e *Exporter) parseInfo(i *io.ReadCloser, ch chan<- prometheus.Metric) error {
+	var version, release_date string
+	s := bufio.NewScanner(*i)
+	for s.Scan() {
+		line := s.Text()
+		if !strings.Contains(line, ":") {
+			continue
+		}
+
+		field := strings.SplitN(line, ": ", 2)
+		switch field[0] {
+		case "Release_date":
+			release_date = field[1]
+		case "Version":
+			version = field[1]
+		}
+	}
+	ch <- prometheus.MustNewConstMetric(haproxyInfo, prometheus.GaugeValue, 1, release_date, version)
```

Comment:
What about not exporting the metric at all in the case of errors? I wouldn't expect to see a metric with empty labels in this case.

Reply:
I thought about that a bit. I was thinking that I didn't want a disappearing metric. But if we can't parse, a flapping value is probably worse.

Reply:
Removed the metric update from the parseInfo function.